### DISCUSSION TITLE
Where do we see the on_dag_run_* listener event logs

### DISCUSSION THREAD

**Discussion by mishrasangeeta87** [apache/airflow#37751] [Q&A]
Hello team,

I've noticed that I can view the on_task_instance_* events log in the Airflow UI. However, I'm encountering an issue where I can't find the on_dag_run_* event logs in my Airflow standalone setup. Could you please guide me to the correct location to access these logs?

**Comment by potiuk** [comment]
look in the logs folder in general.

**Reply by mishrasangeeta87** [reply]
Thank you. Regarding the distributed airflow architecture, which comprises a worker, scheduler, and webserver, if the default logging option is enabled, should we locate all the logs within the base_log_folder directory for these components in order to fetch the on_dag_run_* event listener logs?

**Reply by potiuk** [reply]
Likely

**Reply by mishrasangeeta87** [reply]
Thanks @potiuk . Regarding my earlier query, I didn't find the logs pertaining to the plugin's `on_dag_run*` events within the log directories housed in the specified `base_log_folder` even after conducting a recursive search. Contrarily, these logs were displayed on the console following the execution of the standalone airflow command. I am uncertain whether these logs are persistently stored in a specific location. Any insights on the persistence of these logs would be appreciated.

The console logs looked like this, but wasn't able to find any trace of the text even after doing a recursive search inside the logs directory:
scheduler  | [2024-02-29T12:31:07.797+0530] {events_listener.py:22} INFO - Listener Event==> on_dag_run_running
scheduler  | [2024-02-29T12:31:07.798+0530] {events_listener.py:30} INFO - Observing the DAG with dag_id  test_dag

**Reply by potiuk** [reply]
Airflow has distributed components and different listeners and events are produced in different components.

Standalone airflow is for debuggigng only. By definition it prints logs in the console - as you see here those were printed by scheduler, so (at least theorethically) they should be in the logs of scheduler that are configured by logging as described in https://airflow.apache.org/docs/apache-airflow/stable/administration-and-deployment/logging-monitoring/advanced-logging-configuration.html.

If you can't find them in scheduler - play with those - maybe you will find some more details - you can configure your own loggers etc. If you cannot find out - describe your configuration in details, explain the logs you looked at an open an issue for that. If we have an easily reproducible case of it, possibly somoene will be able to look at it (but also you might try to debug it yourself, looking at what logger configuration you have and figure it out - maybe your airflow deployment has some specific logging configuration that prevents it or maybe you simply have no access to logs of scheduler?

**Reply by mishrasangeeta87** [reply]
Thanks, I will try it out.
